src/python/include/Simplex_tree_interface.h

```diff
@@ -233,6 +233,12 @@ class Simplex_tree_interface : public Simplex_tree<SimplexTreeOptions> {
     auto boundary_srange = Base::boundary_simplex_range(bd_sh);
     return std::make_pair(boundary_srange.begin(), boundary_srange.end());
   }
+
+  Simplex_tree_interface* deserialize_itf(char* buffer, const std::size_t buffer_size) {
+    auto b_ptr = Base::deserialize(buffer, buffer_size);
+    // dynamic_cast should be ok, but compilation fails
+    return reinterpret_cast<Simplex_tree_interface*>(b_ptr);
```

Comment:
I don't think that works. Did you try using extended persistence afterwards? Simplex_tree_interface contains more data than Simplex_tree, interpreting a Simplex_tree* as a Simplex_tree_interface* means you may access memory that is not part of the allocation.

Reply:
This fix was not a smart move :sweat_smile: ... deserialize is now a Simplex_tree method for Simplex_tree_interface to use it. Check the Simplex_tree is empty before calling it. Unitary tests of this feature on [cff6fc2](https://github.com/GUDHI/gudhi-devel/pull/810/commits/cff6fc251be64947b0200a7fc454ae7228170bbb)